### PR TITLE
Merge pose

### PR DESCRIPTION
**Pull request by YHerm**

### CODE REVIEW

**Review by GilStein1 — COMMENTED** [review]
merge non-linear-filters branch, some changes were made

**Comment by Sh-Pe** [comment]
@GilStein1 It's called *not*-linear-filters, my testing branch is *non*-linear-filters-test (both has linear filter). 
Also requires retesting after merging

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -114,7 +115,8 @@ public Robot() {
 			swerve.getConstants().kinematics(),
 			swerve.getModules().getWheelsPositions(0),
 			swerve.getAbsoluteHeading(),
-			PoseEstimatorConstants.DEFAULT_ODOMETRY_STANDARD_DEVIATIONS
+			PoseEstimatorConstants.DEFAULT_ODOMETRY_STANDARD_DEVIATIONS,
+			new VisionDenoiser(20)
```
Comment (GilStein1) [inline src/main/java/frc/robot/Robot.java]: constants?

Comment (Sh-Pe) [reply src/main/java/frc/robot/Robot.java]: I added a constant in the robot template. 
@YHerm you can preform a merge between two different project, consider doing that

File: src/main/java/frc/robot/poseestimator/GBPoseEstimator.java [hunk 2]
```diff
@@ -74,10 +80,29 @@ public ILimelightFilterer getLimelightFilterer() {
...
+			updateGyroOffsetInPose();
+//			Pose2d oldPose = estimatedPose;
+//			if (oldPose == null) {
+//				oldPose = new Pose2d();
+//			}
+//			resetPose(new Pose2d(
+//					oldPose.getX(),
+//					oldPose.getY(),
+//					gyroAngle.plus(headingOffset).rotateBy(Rotation2d.fromDegrees(180))
+//			));
```
Comment (Sh-Pe) [inline src/main/java/frc/robot/poseestimator/GBPoseEstimator.java]: If it's working without it then remove it

**Pull request merged**